### DISCUSSION TITLE
Add feature to reserve informative text of streaming bot messages

### DISCUSSION THREAD

**Discussion by pat-li-ng** [microsoft/teams-sdk#2249] [Ideas / Feature requests]
Based on [Streaming bot messages](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/streaming-ux), we tested this UI by sending chunks of informative text and response text. But seems like the informative text is transient, when start to send response text or entire stream is finished, the informative text and it's UI is disappeared.

For LLM based application, especially agentic LLM app, the informative text will contain massive useful info of what the AI is doing and how. So could we have some APIs to reserve the informative text and control whether it's collapsed by default?

**Comment by corinagum** [comment]
Informative streaming was designed to provide users with insight before response generation. Aside from updates to the AI SDK, adding persistence to this will require support from our Teams client and service teams. Please feel free to file a feature request and I will forward to our PM team for consideration
